File: mobile-client/android/UseCase/Identity/Session/LoadSessionIdentityUseCase.kt
package app.mobiling.client.client.usecase.identity.session

import app.mobiling.client.client.contract.identity.session.SessionIdentityPayload
import app.mobiling.client.client.data.identity.session.SessionIdentityGateway

class LoadSessionIdentityUseCase(
    private val gateway: SessionIdentityGateway,
) {
    suspend operator fun invoke(): SessionIdentityPayload = gateway.loadSessionIdentity()
}

File: client/ios/UseCase/Identity/Session/LoadSessionIdentityUseCase.swift
public struct LoadSessionIdentityUseCase {
    private let gateway: SessionIdentityGateway

    public init(gateway: SessionIdentityGateway) {
        self.gateway = gateway
    }

    public func callAsFunction() async throws -> SessionIdentityPayload {
        try await gateway.loadSessionIdentity()
    }
}
